src/python/com/illinoistriangle/bot/outline_bot.py

```diff
@@ -72,39 +70,81 @@ def lambda_handler(data, context):
   handle_event(slack_event)
 
 
-def handle_event(e):
-  if 'subtype' in e and e['subtype'] == VALUE_TYPE_BOT_MESSAGE:
-    return
+def search_thread_parent_for_urls(channel, thread_ts):
+  """return list of urls from thread's parent post"""
+  #TODO enable scope channels:history
+  response = sc.api_call(
+              'channels.replies',
+              channel=channel,
+              thread_ts=thread_ts,
+              )
+
+  urls = []
+  if response['ok'] == 'true' and 'messages' in response:
+    # from all thread messages, find earliest ts and return the text of that post
+    parent_thread_text = sorted([x['text'] for x in response['messages']], key=lambda x: x['ts'])[0]
+
+    urls = find_urls(parent_thread_text)
+
+  return urls
 
-  if e['type'] != VALUE_TYPE_MESSAGE:
-    return
 
-  if FIELD_TEXT not in e:
-    return
+def event_is_human_thread_reply(event):
+  """return True if the event warrants the bot's response"""
 
-  channel_id = e[FIELD_CHANNEL]
-  if channel_id not in channel_map:
-    return
+  # if a message was posted...
+  if 'type' in event and event['type'] == 'message':
+    # ... by someone other than a bot...
+    if 'subtype' not in event or event['subtype'] != 'bot_message':
+      # ... in a thread...
+      if 'thread_ts' in event:
+        # TODO add channel whitelist
+        # ... in an appropriate channel...
+        if 'channel' in event and event['channel'] in ['C7RLA5HAA', ]:
+          # ... and the bot was called...
+          if 'text' in event and BOT_CALLOUT in event['text']:
+            return True
 
-  if channel_map[channel_id]['name'] not in ['test_x']:
-    return
-  urls = find_urls(e[FIELD_TEXT])
-  logger.info("urls: {}".format(urls))
-  outline_urls = ['https://outline.com/' + u for u in urls]
-  sc.api_call(
-    "chat.postMessage",
-    # TODO: param this.
-    channel=channel_id,
-    text=' '.join(outline_urls)
-  )
+  return False
 
 
+def handle_event(event):
+  # only reply to thread messages, made by humans, that call out the bot
+  if event_is_human_thread_reply(event):
```

Comment:
you want to guard clause here to remove cluster, e.g.:
```
if not event_is_human_thread_reply(event):
  return

urls = search_thread_parent_for_urls(event['thread_ts'], event['thread_ts'])
...
```
this would make the code more clear